code/modules/economy/shippingmarket.dm

```diff
@@ -392,6 +414,20 @@
 
 		return duckets
 
+	proc/appraise_pressure_crystal(var/obj/item/pressure_crystal/pc, var/sell = 0)
+		if (pc.pressure <= 0)
+			return
+		var/index = "[ceil(pc.pressure / PRESSURE_CRYSTAL_SALES_RANGE_LENGTH)]"
```

Comment:
Not really sure what this index system is supposed to be doing, but it's it doesn't seem like it's going to prevent slightly diluting the mix a few times to get different enough values to trick the system. I suggest making it gradually scale by proximity instead, so a lot of similar values will all dilute any more sales in the approximate area.

Reply:
The index takes a measured pressure and converts it into an integer, which represents a range. e.g. pressure of 27.4 gets fit into the range of 25-30. Then if it gets sold, the value of the 25-30 range gets an entry in the sales list as a key, with its value as a multiplier (rand 0.3 to 0.6)
After that, credit value for pressures in that range receives that multiplier. Which stacks multiplicatively with further sales.

It's not perfect: has a hard cutoff, like you say. But I don't have the knowledge to make the system you're describing.